src/plugins/workspace/server/plugin.ts

```diff
@@ -83,8 +88,35 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
     if (isPermissionControlEnabled) {
       this.permissionControl = new SavedObjectsPermissionControl(this.logger);
 
+      this.logger.info('Dynamic application configuration enabled:' + !!applicationConfig);
+      if (!!applicationConfig) {
+        core.http.registerOnPostAuth(async (request, response, toolkit) => {
+          const [coreStart] = await core.getStartServices();
+          const scopeClient = coreStart.opensearch.client.asScoped(request);
+          const configClient = applicationConfig.getConfigurationClient(scopeClient);
+
+          const [adminGroups, adminUsers] = await Promise.all([
+            configClient.getEntityConfig('workspace.dashboardAdmin.groups').catch(() => undefined),
+            configClient.getEntityConfig('workspace.dashboardAdmin.users').catch(() => undefined),
+          ]);
+
+          const isDashboardAdmin = isRequestByDashboardAdmin(
+            request,
+            adminGroups ? [adminGroups] : [],
+            adminUsers ? [adminUsers] : [],
+            this.permissionControl!
```

Comment:
Should we use assert here, `permissionControl`  is inited above and its type is fixed.

Reply:
You are right, the relevant logic has been deleted.